streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -711,22 +736,46 @@ public Subscription subscription(final Set<String> topics) {
                 }
 
                 // finally, encode the assignment before sending back to coordinator
+                final ByteBuffer bb = new AssignmentInfo(minUserMetadataVersion, active, standby, tasksByHostState, 0).encode();
+                assignmentSize = assignmentSize + bb.array().length;
                 assignment.put(consumer, new Assignment(
-                    activePartitions,
-                    new AssignmentInfo(minUserMetadataVersion, active, standby, partitionsByHostState, 0).encode()));
+                        activePartitions, bb));
             }
         }
 
+        log.info("Size of assignment is: " + assignmentSize + " bytes. Total hosts in this assignment are " + clientsMetadata.size());
         return assignment;
     }
 
+    Map<HostInfo, Set<TaskId>> convertTopicPartitionsToTaskIds(final Map<HostInfo, Set<TopicPartition>> partitionsByHostState) {
+        final Map<Integer, InternalTopologyBuilder.TopicsInfo> topicGroups = this.taskManager.builder().topicGroups();
+        final Map<String, Integer> topicToTaskMap = new HashMap<>();
+        for (final Map.Entry<Integer, InternalTopologyBuilder.TopicsInfo> entry : topicGroups.entrySet()) {
+            final Set<String> topicGroupsForHost = new HashSet<>(entry.getValue().sourceTopics);
+            for (final String sourceTopic : topicGroupsForHost) {
```

Comment:
There's no need to copy entry.getValue().sourceTopics into a HashSet just to iterate it. Later down you put individual soureTopic into a topicToTaskMap which already eliminates possible duplicate keys...

Reply:
Yeah, that feels redundant. Addressed the review comments in a push below.